File: sdk-python/examples/contacts/retrieve_payment_method.py
"""Run with: python examples/contacts/retrieve_payment_method.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        method = client.contacts.retrieve_payment_method("cnt_replace_with_real_id")
        if method is None:
            print("no card on file")
            return
        print(f"{method.card.brand} ****{method.card.last4} ({method.status})")
        print(f"  expires:  {method.card.exp_month:02d}/{method.card.exp_year}")
        print(f"  id:       {method.id}")


if __name__ == "__main__":
    main()

File: sdk-node/examples/contacts/retrieve-payment-method.ts
/**
 * Retrieve the saved card on file for a contact (or null when none is saved).
 *
 * Run:
 *   npx tsx examples/contacts/retrieve-payment-method.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const method = await client.contacts.retrievePaymentMethod('cnt_replace_with_real_id')

if (method === null) {
  console.log('no card on file')
} else {
  console.log(`${method.card.brand} ••••${method.card.last4}`)
  console.log(`  expires:  ${String(method.card.expMonth)}/${String(method.card.expYear)}`)
  console.log(`  status:   ${method.status}`)
}
